lib/removeUnusedTechniques.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = removeUnusedTechniques;
+
+function removeUnusedTechniques(gltf, stats) {
+    var usedTechniqueIds = {};
+    var materials = gltf.materials;
+
+    // Build hash of used techniques by iterating through materials
+    if (defined(materials)) {
+        for (var materialId in materials) {
+            if (materials.hasOwnProperty(materialId)) {
+                var id = materials[materialId].technique;
```

Comment:
Careful, `technique` is optional.  We need a `defined` check here.

See https://github.com/KhronosGroup/glTF/blob/master/specification/README.md#materialtechnique